### PR TITLE
Refactored the code that handles Bloop launcher to call it directly.

### PR DESCRIPTION
**Pull request by odisseus**
Also updated to the latest version of Bloop.

Coincidentally, this pull request seems to fix issue 369.

### CODE REVIEW

File: src/core/data.scala [hunk 1]
```diff
@@ -212,20 +213,53 @@ object Compilation {
...
+        charset = StandardCharsets.UTF_8,
+        shell = bloop.launcher.core.Shell.default,
+        nailgunPort = None,
+        startedServer = Promise[Unit](),
+        generateBloopInstallerURL = bloop.launcher.core.Installer.defaultWebsiteURL
+      )
+
+      val launcherStatus = Future{
+        bloopLauncher.runLauncher(
+          bloopVersionToInstall = "1.2.5+271-7c4a6e6a",
```
Comment (propensive) [inline src/core/data.scala]: Can we reuse this snapshot version on the next line without repeating it?

File: src/core/data.scala [hunk 2]
```diff
@@ -212,20 +213,53 @@ object Compilation {
...
+          bloopInstallerURL = bloop.launcher.core.Installer.defaultWebsiteURL("1.2.5+271-7c4a6e6a"),
+          skipBspConnection = false,
+          serverJvmOptions = Nil
+        )
+      }
+
+      val err = new java.io.BufferedReader(new java.io.InputStreamReader(lspErrIn))
       // FIXME: This surely isn't the best way to consume a stream.
       new Thread { override def run(): Unit = while(true) log.println(err.readLine) }.start()
```
Comment (propensive) [inline src/core/data.scala]: Is there a `PipedSomething` you can use to get rid of this line?

Comment (propensive) [reply src/core/data.scala]: Note that it's *essential* that the stream gets consumed, otherwise a buffer fills up and Bloop will block.

**Comment by propensive** [comment]
Why did this get closed?

**Comment by odisseus** [comment]
I have messed up stream handling, so the LSP4J client was waiting for input and never got it.

**Pull request closed without merging**